packages/react/src/hooks/useVoiceAssistant.ts

```diff
@@ -42,8 +43,19 @@ const state_attribute = 'lk.agent.state';
  * @beta
  */
 export function useVoiceAssistant(): VoiceAssistant {
-  const agent = useRemoteParticipants().find((p) => p.kind === ParticipantKind.AGENT);
-  const audioTrack = useParticipantTracks([Track.Source.Microphone], agent?.identity)[0];
+  const agent = useRemoteParticipants().find(
+    (p) => p.kind === ParticipantKind.AGENT && !('lk.publish_on_behalf' in p.attributes),
+  );
+  const worker = useRemoteParticipants().find(
```

Comment:
nit: no need to call `useRemoteParticipants` twice. Storing it once looks a lot cleaner. 
The hook sets up a bunch of internal event listeners, so it's nicer if we can avoid multiple unnecessary calls to it.